arviz/plots/backends/matplotlib/__init__.py

```diff
@@ -52,7 +52,11 @@ def create_axes_grid(length_plotters, rows=1, cols=1, backend_kwargs=None):
 
     backend_kwargs = {**backend_kwarg_defaults(), **backend_kwargs}
 
-    fig, axes = subplots(rows, cols, **backend_kwargs)
+    subplots_kws = {"nrows", "ncols", "sharex", "sharey", "squeeze", "subplot_kw", "gridspec_kw"}
+    fig_kwargs = {key: backend_kwargs[key] for key in backend_kwargs.keys() - subplots_kws}
+    backend_kwargs = {key: backend_kwargs[key] for key in backend_kwargs.keys() & subplots_kws} 
+    fig = Figure(**fig_kwargs)
+    axes = fig.subplots(rows, cols, **backend_kwargs)
```

Comment:
Is this needed to solve the issue? I don't see how this changes the behaviour to users and makes the code more complex and prone to needing maintenance.

Reply:
I think only difference is that now jupyter does not automatically plot the figure.

Everything else should work as is

Reply:
Hmm, maybe my solution does not actually work, will need to check it